libriscan/biblios/static/js/homepage.js

```diff
@@ -0,0 +1,298 @@
+      // Add active styling to clicked tab - matching Start Transcribing button
+      this.classList.add('tab-active', 'btn-primary');
+      this.style.fontWeight = '600';
+      this.style.boxShadow = '0 4px 6px -1px rgba(0, 0, 0, 0.1)';
+      this.style.transform = 'translateY(-2px)';
+      
+      // Hide all panels
+      panels.forEach(panel => panel.classList.add('hidden'));
+      
+      // Show target panel
+      const targetId = this.getAttribute('data-tab');
+      document.getElementById(targetId).classList.remove('hidden');
+      
+      // Save active tab in URL hash
+      window.location.hash = targetId;
+    });
+  });
+  
+  // Restore active tab from URL hash on page load (after pagination)
+  function restoreActiveTab() {
+    const hash = window.location.hash.slice(1); // Remove # from hash
+    if (hash && document.getElementById(hash)) {
+      const tabToActivate = document.querySelector(`[data-tab="${hash}"]`);
+      if (tabToActivate) {
+        tabToActivate.click();
+      }
+    }
+  }
+  
+  // Restore tab on page load
+  restoreActiveTab();
+  
+  // List/Grid toggle buttons
+  const listBtn = document.getElementById('list-view-btn');
+  const gridBtn = document.getElementById('grid-view-btn');
+  
+  if (listBtn && gridBtn) {
+    listBtn.addEventListener('click', function() {
+      listBtn.classList.add('btn-primary');
+      gridBtn.classList.remove('btn-primary');
+    });
+    
+    gridBtn.addEventListener('click', function() {
+      gridBtn.classList.add('btn-primary');
+      listBtn.classList.remove('btn-primary');
+    });
+  }
+  
+  // Search functionality
+  const searchInput = document.getElementById('homepage-search');
+  const searchResults = document.getElementById('homepage-search-results');
+  let searchTimeout;
+  
+  if (searchInput && searchResults) {
+    searchInput.addEventListener('input', function(e) {
+      const query = e.target.value.trim().toLowerCase();
+      
+      // Clear previous timeout
+      clearTimeout(searchTimeout);
+      
+      // Filter the table/grid immediately
+      filterTable(query);
+      
+      // Hide dropdown results if query is empty
+      if (query.length === 0) {
+        searchResults.classList.add('hidden');
+        searchResults.innerHTML = '';
+        return;
+      }
+      
+      // Wait 300ms after user stops typing before showing dropdown
+      searchTimeout = setTimeout(() => {
+        performSearch(query);
+      }, 300);
+    });
+    
+    // Close dropdown results when clicking outside
+    document.addEventListener('click', function(e) {
+      if (!searchInput.contains(e.target) && !searchResults.contains(e.target)) {
+        searchResults.classList.add('hidden');
+      }
+    });
+  }
+  
+  // Filter table rows AND grid cards based on search query
+  function filterTable(query) {
+    const table = document.querySelector('#all-documents table');
+    const gridContainer = document.getElementById('documents-grid');
```

Comment:
Making sure queries are lower case
```suggestion
    const lowerQuery = query.toLowerCase();
    const table = document.querySelector('#all-documents table');
    const gridContainer = document.getElementById('documents-grid');
    let visibleCount = 0;
```